Fixed some minors bugs

pedroaugustobt opened the pull request.

Stexinator commented on module/scripts/combat/defenses.js:
```diff
@@ -135,7 +135,7 @@ function ExecuteDefense(actor, messageId, totalAttack) {
 async function defense(
     actor,
-    { skillName, modifier = 0, stagger = false, block = false },
+    { skillName, modifier = 0, stagger = false, block = false, parry = false },
```
This is unnecessary. See the existing parry block

Stexinator commented on module/scripts/combat/defenses.js:
```diff
@@ -94,7 +94,7 @@ function ExecuteDefense(actor, messageId, totalAttack) {
                     defense(
                         actor,
-                        { skillName: defenseSkill, modifier: -3, stagger: true },
+                        { skillName: defenseSkill, modifier: -3, stagger: true, parry: true },
```
This is unnecessary. See the existing parry block

Stexinator commented on module/scripts/combat/defenses.js:
```diff
@@ -421,6 +422,21 @@ function handleDefenseResults(
             actor.toggleStatusEffect('stun');
         }
+
+        if (parry) {
```
You can add this to the existing parry block

Stexinator commented on module/setup/deprecations.js:
```diff
@@ -27,7 +27,7 @@ async function globalModifiers() {
     let affectedActors = game.actors
         .filter(actor => actor.isOwner)
-        .filter(actor => actor.system.specialSkillModifiers.length > 0);
+        .filter(actor => actor.system.specialSkillModifiers && actor.system.specialSkillModifiers.length > 0);
```
An empty array is a falsy value. So this should not be necessary

pedroaugustobt replied:
> An empty array is a falsy value. So this should not be necessary

This error appears. Is it possible to remove this error log in another way?
![image](https://github.com/user-attachments/assets/484f793a-fd94-45e1-9b29-26b58f7bcd6d)

Stexinator commented on templates/partials/character/tab-inventory.hbs:
```diff
@@ -118,15 +118,20 @@
                                                 <b>{{localize "WITCHER.Weapon.Short.Concealment"}}: </b>
-                                            </label>{{weapon.system.conceal}}
+                                            </label>
+                                            {{#if (eq weapon.system.conceal "T")}}
```
You should use lookups instead. Look at how I changed the hands for weapons

Stexinator commented:
Stating the absence of one thing, does not automatically confirms the presence of another thing

pedroaugustobt commented:
> Stating the absence of one thing, does not automatically confirms the presence of another thing.
> 
> Else you would also need to remove the reliability damage from blocking, due to the text saying parrying. As far as I can see they made a major fumble when splitting block and parry and did not clean up the texts

yes, it's clearer to me now, I asked other GMs and it seems that the damage really only applies to blocking. Sorry Stex, i'll remove the commit

pedroaugustobt commented:
@Stexinator I removed the parry commit and fixed the deprecation log.
If there are any more pending issues or something that is wrong that I need to change, just tell me here!

Stexinator commented on module/scripts/rolls/extendedRoll.js:
```diff
@@ -10,6 +10,7 @@ export async function extendedRoll(rollFormula, messageData, config, flags = [])
 
     //crit/fumble calculation
     if (config.showCrit && (isCrit(roll) || isFumble(roll))) {
+        await game.dice3d?.showForRoll(roll, game.user, true);
```
So this roll will only show if there is a crit or fumble?

pedroaugustobt replied:
the roll will happen normally if it is not a critical or fumble.

Stexinator replied:
Yes, and you only show the roll with 3d dice when it is a crit or fumble. Read the line above

Stexinator commented on templates/partials/character/tab-inventory.hbs:
```diff
@@ -107,18 +107,17 @@
-                                            </div>
+
+                                            </div> 
+
```
Why did you add empty lines?

pedroaugustobt commented:
> Why did you close it? Dont you want to have these changes in the system?

Hey Stex, I'll redo them here, I updated my repository with the latest update (v12.15.3). I'm working on it now!
